### ISSUE TITLE
got stuck when training a smaller dataset

### ISSUE THREAD

**Issue by TraceOnJYP** [facebookresearch/deepmask#36]
I want to just have a try training in a smaller dataset(about 150 image files, and I also generate my own annotation json file). when running train.lua, the program got stuck in "start training". I print the running information and find it was stuck at "dojob()" in dataloader.lua where n=7. what's wrong?

**Comment by poliveirap** [comment]
What is n? number of threads?
At least when training on the COCO dataset (which is much bigger than the one you are using), there makes no difference in using 2 or more threads. 
Maybe try to reduce it to maybe 3 or 4 to see if it works? (it should not make any difference in terms on training speed)

**Comment by TraceOnJYP** [comment]
@poliveirap n is the loop number which is returned when dataloader:run() is called. And I also tried 3 or other nthreads argument, it also got stuck but not the same n.

**Comment by TraceOnJYP** [comment]
@poliveirap I have a question, in "DataSampler.lua":
`  while not ann or ann.iscrowd == 1 or ann.area < 100 or ann.bbox[3] < 5
    or ann.bbox[4] < 5 do
      local catId = self.catIds[cat]
      local annIds = self.coco:getAnnIds({catId=catId})
      local annid = annIds[torch.random(annIds:size(1))]
      ann = self.coco:loadAnns(annid)[1]
  end
  local bbox = self:jitterBox(ann.bbox)
  local imgName = self.coco:loadImgs(ann.image_id)[1].file_name`

why should the masksampling img be always iscrowd=0 that is a polygon segmentation?

**Comment by poliveirap** [comment]
we train the mask branch by always considering an object fully contained in the patch and roughly located in center and in the right scale (see the paper for more detail). 
The object labeled with iscrowd do not really satisfy this constraint (if I remember, it is a bunch of objects close to each other all labeled as one big crowd), thus not in agreement with the training data definition.

**Comment by tongerpeng** [comment]
@TraceOnJYP Hi，I want to know how to train model with my own images and how to generate annotation json file????

**Comment by tongerpeng** [comment]
@TraceOnJYP And I don't know where to place my own images and annntation json file

**Comment by chrisckr** [comment]
@TraceOnJYP Did you find a solution to your problem? I think I'm running into the same thing: COCO training is working fine, now I'm training on my own dataset, the json > t7 convertion works, but then it is stuck at "start training" (CPU is still at 100% on a P2 AWS instance, but it should finish after a couple of minutes in comparsion to COCO). Set max epochs to 2 but the same result. Small dataset, only one category (hardcoded value 80 in Datasampler switched to 1), the annotation files are correct (checked 10 times). Any tips how to debug this or what could be the problem? Thanks.

**Comment by nirmaljith** [comment]
@TraceOnJYP I am also facing the same issue #98. Did you able to run the code ?

**Comment by nirmaljith** [comment]
I got a fix for the issue #98.
